dashboard/modules/metrics/install_and_start_prometheus.py

```diff
@@ -0,0 +1,192 @@
+    except Exception as e:
+        logging.error(f"Failed to start Prometheus: {e}")
+        return None
+
+
+def print_shutdown_message(process_id):
+    message = (
+        f"Prometheus is running with PID {process_id}.\n"
+        "To stop Prometheus, use the command: "
+        f"'kill {process_id}', or if you need to force stop, "
+        f"use 'kill -9 {process_id}'."
+    )
+    print(message)
+
+    debug_message = (
+        "To list all processes running Prometheus, use the command: "
+        "'ps aux | grep prometheus'."
+    )
+    print(debug_message)
+
+
+def get_latest_prometheus_version():
+    url = "https://api.github.com/repos/prometheus/prometheus/releases/latest"
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        data = response.json()
+        # Remove the leading 'v' from the version number
+        return data["tag_name"].lstrip("v")
+    except requests.RequestException as e:
+        logging.error(f"Error fetching latest Prometheus version: {e}")
+        return None
+
+
+def get_prometheus_filename(os_type=None, architecture=None, prometheus_version=None):
+    if os_type is None or architecture is None:
+        os_type, architecture = get_system_info()
+
+    if prometheus_version is None:
+        prometheus_version = get_latest_prometheus_version()
+        if prometheus_version is None:
+            logging.warning(
+                "Failed to retrieve the latest Prometheus version. Falling "
+                f"back to {FALLBACK_PROMETHEUS_VERSION}."
+            )
+            # Fall back to a hardcoded version
+            prometheus_version = FALLBACK_PROMETHEUS_VERSION
+
+    return (
+        f"prometheus-{prometheus_version}.{os_type}-{architecture}.tar.gz",
+        prometheus_version,
+    )
+
+
+def get_prometheus_download_url(
+    os_type=None, architecture=None, prometheus_version=None
+):
+    file_name, prometheus_version = get_prometheus_filename(
+        os_type, architecture, prometheus_version
+    )
+    return (
+        "https://github.com/prometheus/prometheus/releases/"
+        f"download/v{prometheus_version}/{file_name}"
+    )
+
+
+def download_prometheus(os_type=None, architecture=None, prometheus_version=None):
+    file_name, _ = get_prometheus_filename(os_type, architecture, prometheus_version)
+    download_url = get_prometheus_download_url(
+        os_type, architecture, prometheus_version
+    )
+
+    return download_file(download_url, file_name), file_name
+
+
+def main():
+    logging.warning("This script is not intended for production use.")
+
+    downloaded, file_name = download_prometheus()
+    if not downloaded:
+        logging.error("Failed to download Prometheus.")
+        sys.exit(1)
+
+    # TODO: Verify the checksum of the downloaded file
+
+    if not install_prometheus(file_name):
+        logging.error("Installation failed.")
+        sys.exit(1)
+
+    # TODO: Add a check to see if Prometheus is already running
```

Comment:
let's do this TODO. Maybe just a simple check of the prometheus port?

Reply:
Will file a followup PR for this. I remember Prometheus prints a reasonable error if the port is already in use, so it should be okay.